pytorch_lightning/loops/base.py

```diff
@@ -59,6 +64,10 @@ def skip(self) -> bool:
     def connect(self, trainer: 'pl.Trainer', *args: Any, **kwargs: Any) -> None:
         """Connects Loop with all the necessary things like connectors and accelerators."""
         # TODO(@justusschock): Make the trainer a weakref/proxy
+        if not isinstance(trainer, pl.Trainer):
+            raise MisconfigurationException(
+                f"Loop {self.__class__.__name__} should be connected to a :class:`~pytorch_lightning.Trainer` instance."
```

Comment:
This is not a rendered doc

```suggestion
                f"Loop {self.__class__.__name__} should be connected to a `Trainer`, found: {trainer}."
```

Reply:
Removed ! @awaelchli I will let you explore the corner case there.

Reply:
Done !